django/core/management/__init__.py

```diff
@@ -368,8 +368,9 @@ def execute(self):
         try:
             settings.INSTALLED_APPS
         except ImproperlyConfigured as exc:
-            self.settings_exception = exc
-        except ImportError as exc:
+            # The following commands can be run even without a valid settings file configured
+            if subcommand not in {'startproject', 'startapp', 'makemessages'}:
```

Comment:
Seems wrong to hard code this list when Django fully supports user-written commands.

Reply:
Hi, If the settings module cannot be imported then user written commands will not be available for execution. We only have to care about core commands in this exception block. That said, I do agree that this is pretty bad code. I will pick it up later if no one else gets to it first. Thanks! :)